File: apple_division.py

def min_diff(sorted_array):
    l = len(sorted_array)
    sum1 = sorted_array[l-1]
    sum2 = sum(sorted_array) - sum1
    print("s1 = %d" %sum1)
    print("s2 = %d" %sum2)

    d1 = 10000000000
    d2 = abs(sum2 - sum1)
    print(d1, d2)

    i = 0

    while(d2<d1):
        d1 = d2

        sum1 += sorted_array[i]
        sum2 -= sorted_array[i]
        print("s1 = %d" %sum1)
        print("s2 = %d" %sum2)

        d2 = abs(sum2 - sum1)
        print(d1, d2)
        i+=1

    return d1



if __name__ == "__main__":

    arr = [1, 3, 4, 5, 7]
    print(min_diff(arr))

File: apple_division.cpp
#include <iostream>
#include <algorithm>
#include <cstdlib>
using namespace std;


int div(int nums[], int count){
    int max_num = *max_element(nums, nums+count);

    int min_num = *min_element(nums, nums+count);

    // cout<<max_num<<" "<<min_num<<endl;

    int sum1 = max_num+min_num;
    int sum2 = 0;
    for (int i =0;i<count;i++){
        if(nums[i]!=max_num && nums[i]!=min_num){
            sum2+=nums[i];
        }
    }

    int d1 = abs(sum1 - sum2);

    cout<<d1<<endl;
    return d1;
}

int main(){


    int count;
    cin>>count;

    int nums[count];

    for (int i = 0 ; i < count ; i++){
        cin>>nums[i];
    }

    cout<<div(nums, count);


    // d1, d2
    // while(d2<d1){
        // 
    //}





    return 0;
}
